mitmproxy/proxy/layers/http/__init__.py

```diff
@@ -756,6 +757,15 @@ def handle_connect_finish(self):
                 self.context.client,
             )
             yield SendHttp(ResponseEndOfMessage(self.stream_id), self.context.client)
+        elif self.flow.response.status_code != 407:
```

Comment:
Replacing this condition by an `else` fails this test:
https://github.com/mitmproxy/mitmproxy/blob/968a169077869182414b202520933408dd0a7414/test/mitmproxy/proxy/layers/http/test_http.py#L1636

It seems that we need to send a response to the client after a `connect error` and receive a request from the client in case the status_code is 407, how can we handle such a flow here?